cfe_internal/enterprise/mission_portal.cf

```diff
@@ -120,7 +120,8 @@ bundle agent mission_portal_apache_from_stage(config, staged_config)
 
     "$(config)"
       copy_from => local_dcp( $(staged_config) ),
-      if => returnszero("$(validate_config) > /dev/null 2>&1 ", "useshell"),
+      if => and( "apache_stop_after_new_staged_config_repaired",
```

Comment:
This might be another existing issue, but if I delete the config file httpd.conf it doesn't get restored with this change and maybe wouldn't without this change. My hack which seems to work is to add a case where the config file is zero length due to `create => "true"` above in this policy.

```
    "$(config)"
      copy_from => local_dcp( $(staged_config) ),
# if config is zero size it probably was created in a previous promise
if => or( not(filesize("$(config)")),
       and( "apache_stop_after_new_staged_config_repaired",
                 returnszero("$(validate_config) > /dev/null 2>&1 ", "useshell"))),
      classes => results("bundle", "mission_portal_apache_config"),
      comment => "We make sure that the deployed config is a copy of the staged
                  config if the staged config passes a syntax check. We redirect
                  the output to keep policy runs clean and not generate lots of
                  unnecessary email.";
```

Reply:
Doing some testing I am having a hard time getting this to work. I don't think we need to address re-building httpd.conf at this time. I'll log another ticket for that.